fastf1/core.py

```diff
@@ -2189,46 +2238,49 @@ def pick_teams(self, names):
         """
         return self[self['Team'].isin(names)]
 
-    def pick_fastest(self, only_by_time=False):
+    def pick_fastest(self, round="all"):
         """Return the lap with the fastest lap time.
 
-        This method will by default return the quickest lap out of self, that
-        is also marked as personal best lap of a driver.
-
-        If the quickest lap by lap time is not marked as personal best, this
-        means that it was not counted. This can be the case for example, if
-        the driver exceeded track limits and the lap time was deleted.
-
-        If no lap is marked as personal best lap or self contains no laps,
-        an empty Lap object will be returned.
-
-        The check for personal best lap can be disabled, so that any quickest
-        lap will be returned.
+        This method will by default return the quickest lap out of self that is valid.
+        Maybe there is a fastest lap made by this driver, but it was deleted due to track limits
+        so just the oficially fastest one is returned.
 
         Args:
-            only_by_time (bool): Ignore whether any laps are marked as
-                personal best laps and simply return the lap that has the
-                lowest lap time.
+            round(String): By default "all" rounds are considered, so it returns the fastest oficial lap
+                of the whole qualifying session. 
+                By setting round to "Q1" the fastet oficial lap of Q1 will be returned
+                By setting round to "Q2" the fastet oficial lap of Q2 will be returned
+                By setting round to "Q3" the fastet oficial lap of Q3 will be returned
+
+                If the driver, did not take part in the round selected, an empty instance of lap will be returned
 
         Returns:
             instance of :class:`Lap`
         """
-        if only_by_time:
-            laps = self  # all laps
-        else:
-            # select only laps marked as personal fastest
-            laps = self.loc[self['IsPersonalBest'] == True]  # noqa: E712 comparison with True
-
+        
+        laps = self  # all laps
         if not laps.size:
             return Lap(index=self.columns)
 
-        lap = laps.loc[laps['LapTime'].idxmin()]
-        if isinstance(lap, pd.DataFrame):
-            # More laps, same time
-            lap = lap.iloc[0]  # take first clocked
+        min= timedelta(hours=0, minutes=99, seconds=99)
+        i=0
+        lap=Lap(index=self.columns)
+        for i in range(len(laps)):
+            lapRound=laps.iloc[i].get_quali_round()
+            if(lapRound!="-1" and(round=="all" or lapRound==round)):
+                if(laps.iloc[i]['LapTime']<min):
+                    if(laps.iloc[i].isValid()):
+                        lap=laps.iloc[i]
+                        min=laps.iloc[i]['LapTime']
+                    else:
+                        print("fastest was "+str(laps.iloc[i]['LapTime'])+" but was deleted")
```

Comment:
In general, the `print` command should never be used in a module like FastF1. Status values should be returned as variables. The users can print those themselves or use them in any other way,